pandas/tests/io/test_parquet.py

```diff
@@ -110,48 +110,72 @@ def df_full():
                                pd.Timestamp('20130103')]})
 
 
-def test_invalid_engine(df_compat):
+def check_round_trip(df, engine=None, path=None,
+                     write_kwargs=None, read_kwargs=None,
+                     expected=None, check_names=True):
 
-    with pytest.raises(ValueError):
-        df_compat.to_parquet('foo', 'bar')
+    if write_kwargs is None:
+        write_kwargs = {'compression': None}
 
+    if read_kwargs is None:
+        read_kwargs = {}
 
-def test_options_py(df_compat, pa):
-    # use the set option
+    if engine:
```

Comment:
is there a case where engine is not required?

Reply:
Yes, mostly when testing for "defaults" handling (i.e. default engine, or engine being configured via `option_context`. Please see here https://github.com/pandas-dev/pandas/pull/19332/files#diff-539eb41a29c68cc27818db4b480c2fd6R163